src/reference/asciidoc/router.adoc

```diff
@@ -627,6 +656,35 @@ However, in cases where the header value is mapped to a channel name but the cha
 IMPORTANT: With Spring Integration 2.1 the attribute was changed from `ignore-channel-name-resolution-failures` to `resolution-required`.
 Attribute `resolution-required` will default to `true`.
 
+The equivalent router, using Java configuration:
+
+[source, java]
+----
+@ServiceActivator(inputChannel = "routingChannel")
+@Bean
+public HeaderValueRouter router() {
+    HeaderValueRouter router = new HeaderValueRouter("testHeader");
+    router.setChannelMapping("someHeaderValue", "channelA");
+    router.setChannelMapping("someOtherHeaderValue", "channelB");
+    return router;
+}
+----
+
+and using the Java DSL:
+
+[source, java]
+----
+@Bean
+public IntegrationFlow routerFlow() {
+    return IntegrationFlows.from("routingChannel")
+            .route(new HeaderValueRouter("testHeader"), m -> m
```

Comment:
We can use, but that is different method based on the `AbstractMessageRouter` argument.

We provide choices for preferences:
1. To use fluent method chain API, which let you type flow quickly and straightforward.
2. To declare separate definition of the canonical component via its setters and then lightweight flow.